pkg/operator/controller/status.go

```diff
@@ -41,10 +40,8 @@ func (r *reconciler) syncOperatorStatus() error {
 		return fmt.Errorf("failed to get operator state: %v", err)
 	}
 
-	oldConditions := co.Status.Conditions
-	co.Status.Conditions = computeStatusConditions(oldConditions, ns, ingresses, deployments)
-
-	oldRelatedObjects := co.Status.RelatedObjects
+	oldStatus := *co.Status.DeepCopy()
```

Comment:
do we need explicit dereference here? (*oc).Status.DeepCopy()

Reply:
`DeepCopy` returns a pointer, so we need to dereference its return value either here or when we pass it to `statusesEqual` further down.